src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -232,33 +231,29 @@ void JNI_OnUnload(JavaVM * jvm, void * reserved)
 
 JNI_METHOD(jlong, newDeviceController)(JNIEnv * env, jobject self)
 {
-    CHIP_ERROR err                          = CHIP_NO_ERROR;
-    ChipDeviceController * deviceController = NULL;
-    long result                             = 0;
+    CHIP_ERROR err                           = CHIP_NO_ERROR;
+    AndroidDeviceControllerWrapper * wrapper = NULL;
+    long result                              = 0;
 
     ChipLogProgress(Controller, "newDeviceController() called");
 
-    deviceController = new ChipDeviceController();
-    VerifyOrExit(deviceController != NULL, err = CHIP_ERROR_NO_MEMORY);
-
-    err = deviceController->Init(kLocalDeviceId, &sSystemLayer, &sInetLayer, &sDevicePairingDelegate);
+    wrapper = AndroidDeviceControllerWrapper::AllocateNew(kLocalDeviceId, &sSystemLayer, &sInetLayer, &err);
     SuccessOrExit(err);
 
-    deviceController->AppState = (void *) env->NewGlobalRef(self);
-
-    result = (long) deviceController;
+    wrapper->Controller()->AppState = (void *) env->NewGlobalRef(self);
+    result                          = wrapper->ToJNIHandle();
 
 exit:
     if (err != CHIP_NO_ERROR)
     {
-        if (deviceController != NULL)
+        result = 0;
```

Comment:
nit: `result` is already `0` if you reach this, or did I misread the code ?

Reply:
I maded the change. Having a specific set may make it clearer that 0 is returned on any error. Then again, I think we ThrowError anyway so result does not seem to really matter anyway.